pilot/cmd/pilot-agent/status/server.go

```diff
@@ -870,6 +875,12 @@ func notifyExit() {
 	}
 }
 
+// errorExit exits with error code -1, before that it will notify envoy to quit as well.
+func errorExit() {
```

Comment:
I don't love the hard Exit() here... but not sure a better option since currently notifyExit is sending a signal.

We could make it work and improve the old flow by passing through some channel to cancel probably